src/modules/job-manager/submit.c

```diff
@@ -104,11 +104,26 @@ static int submit_job (struct job_manager *ctx,
         set_errorf (errors, job->id, "error posting submit event");
         goto error;
     }
+    /* Post the validate event.
+     */
+    if (event_job_post_pack (ctx->event, job, "validate", 0, NULL) < 0) {
+        set_errorf (errors, job->id, "error posting validate event");
+        goto error_post_invalid;
+    }
     if ((job->flags & FLUX_JOB_WAITABLE))
         wait_notify_active (ctx->wait, job);
     if (ctx->max_jobid < job->id)
         ctx->max_jobid = job->id;
     return 0;
+error_post_invalid:
+    /* Let journal consumers know this job did not pass validation and
+     * all data concerning it should be expunged.
+     */
+    (void)event_job_post_pack (ctx->event,
+                               job,
+                               "invalidate",
+                               EVENT_NO_COMMIT,
+                               NULL);
```

Comment:
I might not be understanding the `invalidate` event, but should this event also be posted if a `job.validate` callback fails or  the `jobtap_check_dependencies()` function fails? Currently, this event is only posted if `event_job_post_pack ()` of the `valid` event fails.

Reply:
Ugh yeah, it should _primarily_ be posted in the validation failure case.   When the job fails, the ingest module removes the job directory (including the eventlog) but "invalid" is needed to tell journal consumers that the job is DOA.   Big error, thank you!